hadoop-hdds/rocks-native/src/main/java/org/apache/hadoop/hdds/utils/db/managed/ManagedSSTDumpIterator.java

```diff
@@ -244,40 +245,46 @@ protected void finalize() throws Throwable {
    * Class containing Parsed KeyValue Record from Sst Dumptool output.
    */
   public static final class KeyValue {
-    private String key;
-    private Integer sequence;
-    private Integer type;
 
-    private String value;
+    private final byte[] key;
+    private final UnsignedLong sequence;
+    private final Integer type;
+    private final byte[] value;
 
-    private KeyValue(String key, String sequence, String type,
-                     String value) {
+    private KeyValue(byte[] key, UnsignedLong sequence, Integer type,
+             byte[] value) {
       this.key = key;
-      this.sequence = Integer.valueOf(sequence);
-      this.type = Integer.valueOf(type);
+      this.sequence = sequence;
+      this.type = type;
       this.value = value;
     }
 
-    public String getKey() {
+    @SuppressFBWarnings("EI_EXPOSE_REP")
+    public byte[] getKey() {
       return key;
     }
 
-    public Integer getSequence() {
+    public UnsignedLong getSequence() {
       return sequence;
     }
 
     public Integer getType() {
       return type;
     }
 
-    public String getValue() {
+    @SuppressFBWarnings("EI_EXPOSE_REP")
+    public byte[] getValue() {
       return value;
     }
 
     @Override
     public String toString() {
-      return "KeyValue{" + "key='" + key + '\'' + ", sequence=" + sequence +
-          ", type=" + type + ", value='" + value + '\'' + '}';
+      return "KeyValue{" +
+          "key=" + StringUtils.bytes2String(key) +
```

Comment:
Can `key` and `value` be null? If so, `StringUtils.bytes2String()` will throw NPE if I'm not wrong.

Reply:
no it cannot be null. We throw illegalstateexception even before it comes here.

Reply:
done